packages/samples/stencil/src/components/app-root/app-root.tsx

```diff
@@ -1,30 +1,60 @@
-import {Component, h, Fragment} from '@stencil/core';
-import {createRouter, Route} from 'stencil-router-v2';
-import {Header} from '../header/header';
+import {Component, h} from '@stencil/core';
+import {createRouter, href, Route} from 'stencil-router-v2';
+import {
+  getSampleSearchEngineConfiguration,
+  SearchEngine,
+} from '@coveo/atomic/headless';
 
 const Router = createRouter();
+const homePath = '/';
+const searchPath = '/search';
 
 @Component({
   tag: 'app-root',
   shadow: false,
 })
 export class AppRoot {
-  render() {
+  private engine?: SearchEngine;
+  public async componentDidLoad() {
+    const searchInterface: HTMLAtomicSearchInterfaceElement =
+      document.querySelector('atomic-search-interface')!;
+
+    await searchInterface.initialize(getSampleSearchEngineConfiguration());
+    this.engine = searchInterface.engine!;
+
+    if (Router.activePath === searchPath) {
+      searchInterface.executeFirstSearch();
+    }
+  }
+
+  public render() {
     return (
-      <Router.Switch>
-        <Route
-          path="/"
-          render={() => (
-            <Fragment>
-              <Header>
-                <standalone-search-box router={Router}></standalone-search-box>
-              </Header>
-              <h1>Home</h1>
-            </Fragment>
-          )}
-        />
-        <Route path="/search" render={() => <search-page></search-page>} />
-      </Router.Switch>
+      <atomic-search-interface>
```

Comment:
Along with the standalone search box (switching the redirection-url) this is the best user experience with the Stencil Router, maybe because the search box won't fully be recreated. I found a few bugs inside Atomic.

In the future we should improve the atomic-external or the search box so it's able to dump its engine and hook to a new one. 🤞